backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/DatabaseDocumentNumberRepository.java

```diff
@@ -1,15 +1,13 @@
 package de.bund.digitalservice.ris.caselaw.adapter.database.jpa;
 
-import jakarta.persistence.LockModeType;
+import java.time.Year;
 import java.util.Optional;
-import org.jetbrains.annotations.NotNull;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.data.jpa.repository.Lock;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface DatabaseDocumentNumberRepository extends JpaRepository<DocumentNumberDTO, String> {
-  @NotNull
-  @Lock(LockModeType.PESSIMISTIC_WRITE)
```

Comment:
I think this is something we need also at the new function

Reply:
The idea behind this is to allow only one thread to get a document number at a time. All other threads have to wait until the first thread is ready. If we don't do this two thread can get the same document number and the unique constrain will reject the second one to generate a new documentation unit.